### DISCUSSION TITLE
Sensor seem to go inactive unless I am logged in to HA

### DISCUSSION THREAD

**Discussion by usbrit** [hultenvp/solis-sensor#144] [Q&A]
![image](https://user-images.githubusercontent.com/53425475/186737036-a7d355ba-267b-4b53-85e0-e7085f8c6a33.png)
 Hi,
I installed Solis Sensor on the 23 Aug and was very impressed with what has been created. However, after a few hours the data seems to stay static.

Logged on to HA again yesterday, and the again today. As you can see Battery Power remaining seems to stay static when I am not logged in. 

Is there a refresh setting to keep this updated? I am trying to use Node Red to alert me when battery hits 99% so devices in the house can then be activated on "free" power.

Thanks
David

**Comment by hultenvp** [comment]
Hi @usbrit,

No, there's no refresh setting. The integration gets latest data from the server every 2 minutes and should never stop collecting.
Do you see the same issue with other sensors or do they continue collecting data?
I do assume the graph on the server shows the real battery power and not the same flat line.

**Reply by usbrit** [reply]
Thanks for the quick reply. Unfortunately Ginlong does not provide a battery plot, only an instantaneous value. But I know from checking the value through the day using their mobile app that at 14.00 today my batteries were full. 
I get the same issue with all other sensors I think. Here's the plots from today from my energy dashboard (not configured correctly yet (see my other question about solar output)
HA data:
![image](https://user-images.githubusercontent.com/53425475/186754980-9fcf3eaf-f435-4604-b753-8d73769d4f3e.png)
Solis Cloud data:
![image](https://user-images.githubusercontent.com/53425475/186755273-799ab468-642a-465a-9713-09aa599a935d.png)

**Reply by hultenvp** [reply]
From the graphs it supports only energy to battery is affected. All other sensors are behaving nicely.
Not sure if something is really wrong in the bottom graph from an integration perspective, but the behaviour looks odd. It looks like between 11 and 14 power is taken from grid to charge the battery and it's charging with 2.5 kW. And later, between 16 and 18 it actually draws from the battery and sends to grid. But since you did not yet add solar in the config I'm not sure if HA calculates the values correctly or whether this is correct.
Do you use ginlong.com or soliscloud.com?

**Reply by usbrit** [reply]
Soliscloud through API key.

But I don't agree the graphs match:
![image](https://user-images.githubusercontent.com/53425475/186759742-de663ccd-7e2b-4d2c-ad54-0ecbc18f73a3.png) 
the blue bar is grid energy used and shows 0 from 5.00 till 18.00

On the soliscloud graph peaks at 11.00 till 14.30. and has very little between 14.30 and the end of plot
And no other values are shown between 5.00 till 18.00 either

**Reply by hultenvp** [reply]
I was focused on the soliscloud graphs, was having trouble understanding what I was seeing there. 
The energy dash indeed is a mess.
Can you make a screenshot of a couple of solis sensors in history?
It would help me establish if only battery sensor is suffering or all sensors fail to retrieve data. Something like this:
<img width="946" alt="image" src="https://user-images.githubusercontent.com/61835400/186765662-b39eb37c-94ec-4fec-b173-689256840609.png">

**Reply by usbrit** [reply]
![image](https://user-images.githubusercontent.com/53425475/186767503-80b6923c-9e7d-493f-bab9-4db1070e72e4.png)

Looks like all sensors just stayed at one value until I logged on

![image](https://user-images.githubusercontent.com/53425475/186767826-16b6d150-3551-4b00-8b3d-ff52accfea9a.png)

**Comment by hultenvp** [comment]
I'm really suspecting that your setup goes to sleep or suspend and just stops acquiring.
What does your ha setup look like?

**Reply by usbrit** [reply]
Worked all day today and just need to sort out "power to grid" not matching Solids data (see my other question.

Not sure what the issue was before, but happy now. Will let you know if I need more help with the sleep issue. Thanks
